// @astii/dialog/dist/esm/Dialog/index.js
import React, { useMemo } from "react";
import Mask from "./Mask";
import Wrap from "./Wrap";
import Icon from "../Icon";
import Draggable from "../Draggable";
import Collapse from "./Collapse";
import styles from "./index.module.less";
var Dialog = function Dialog(_ref) {
  var open = _ref.open,
    _ref$title = _ref.title,
    title = _ref$title === void 0 ? "Dialog" : _ref$title,
    _ref$width = _ref.width,
    width = _ref$width === void 0 ? 1024 : _ref$width,
    children = _ref.children,
    footer = _ref.footer,
    draggable = _ref.draggable,
    inside = _ref.inside,
    maskClosable = _ref.maskClosable,
    contentClassName = _ref.contentClassName,
    onClose = _ref.onClose;
  var modal = useMemo(function () {
    var modal = /*#__PURE__*/React.createElement("div", {
      className: styles.dialog,
      style: {
        width: width
      }
    }, /*#__PURE__*/React.createElement("div", {
      "data-handler": true,
      className: styles["dialog-header"]
    }, title), /*#__PURE__*/React.createElement("div", {
      className: "".concat(styles["dialog-content"], " ").concat(contentClassName !== null && contentClassName !== void 0 ? contentClassName : "")
    }, children), /*#__PURE__*/React.createElement(Icon, {
      name: "close",
      className: styles.close,
      onClick: onClose
    }), footer && /*#__PURE__*/React.createElement(Collapse, null, footer));
    return draggable ? /*#__PURE__*/React.createElement(Draggable, {
      inside: inside
    }, modal) : modal;
  }, [draggable, inside]);
  return open ? /*#__PURE__*/React.createElement(Mask, {
    onClick: maskClosable ? onClose : function () {}
  }, /*#__PURE__*/React.createElement(Wrap, null, modal)) : null;
};
Dialog.Icon = Icon;
export default Dialog;
